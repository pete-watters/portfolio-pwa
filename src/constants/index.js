export default {
  ROUTES: {
    HOME: '/',
    ABOUT: '/about',
    BLOG: '/blog',
  },
  SOCIAL: {
    GITHUB: 'https://github.com/pete-watters',
    LINKEDIN: 'https://www.linkedin.com/in/pete-watters/',
    STACKOVERFLOW: 'https://stackoverflow.com/users/1365580/peadar',
    TWITTER: 'https://twitter.com/watters_pete',
    MEDIUM: 'https://medium.com/@pete_watters',
    INSTAGRAM: 'https://www.instagram.com/watterspete/',
  },
};
