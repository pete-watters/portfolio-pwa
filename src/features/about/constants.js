export const VIEWS = {
  ADVENTURES: 'Adventures',
  CONTACT: 'Contact',
  STORY: 'Story',
  WORK: 'Work'
}