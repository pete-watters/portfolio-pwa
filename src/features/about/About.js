import React, { PureComponent } from 'react';
import 'react-responsive-carousel/lib/styles/carousel.min.css';
import Page from 'components/Page';
import { VIEWS } from './constants';
import { getAboutComponent } from './helpers';

class About extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      view: null,
    };

    this.switchView = this.switchView.bind(this);
  }

  switchView(view) {
    this.setState({ view });
  }

  render() {
    const { view } = this.state;
    const ENTER_KEY_CODE = 13;
    return (
      <Page>
        <nav>
          <ul>
            {/* FIXME refactor this - abstract to clickable div element component */}
            <li><div tabIndex="0" role="button" onClick={() => this.switchView(VIEWS.STORY)} onKeyPress={e => e.keyCode === ENTER_KEY_CODE && this.switchView(VIEWS.STORY)}><h4>{VIEWS.STORY}</h4></div></li>
            <li><div tabIndex="0" role="button" onClick={() => this.switchView(VIEWS.WORK)} onKeyPress={e => e.keyCode === ENTER_KEY_CODE && this.switchView(VIEWS.STORY)}><h4>{VIEWS.WORK}</h4></div></li>
            <li><div tabIndex="0" role="button" onClick={() => this.switchView(VIEWS.CONTACT)} onKeyPress={e => e.keyCode === ENTER_KEY_CODE && this.switchView(VIEWS.STORY)}><h4>{VIEWS.CONTACT}</h4></div></li>
            <li><div tabIndex="0" role="button" onClick={() => this.switchView(VIEWS.ADVENTURES)} onKeyPress={e => e.keyCode === ENTER_KEY_CODE && this.switchView(VIEWS.STORY)}><h4>{VIEWS.ADVENTURES}</h4></div></li>
          </ul>
        </nav>
        {getAboutComponent(view)}
      </Page>);
  }
}

export default About;
