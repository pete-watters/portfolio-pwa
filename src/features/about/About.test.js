/* eslint-disable no-magic-numbers */
import React from 'react';
import { shallow } from 'enzyme';
import About from './About';
import { VIEWS } from './constants';

const component = shallow(<About />);

describe('About', () => {
  it('should render the component correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('renders the correct text', () => {
    expect(component.find('h4').at(0).text()).toEqual(VIEWS.STORY);
    expect(component.find('h4').at(1).text()).toEqual(VIEWS.WORK);
    expect(component.find('h4').at(2).text()).toEqual(VIEWS.CONTACT);
    expect(component.find('h4').at(3).text()).toEqual(VIEWS.ADVENTURES);
  });
});
