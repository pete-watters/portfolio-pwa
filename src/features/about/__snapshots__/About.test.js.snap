// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render the component correctly 1`] = `
<Memo(Page)>
  <nav>
    <ul>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <h4>
            Story
          </h4>
        </div>
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <h4>
            Work
          </h4>
        </div>
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <h4>
            Contact
          </h4>
        </div>
      </li>
      <li>
        <div
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <h4>
            Adventures
          </h4>
        </div>
      </li>
    </ul>
  </nav>
  <Story />
</Memo(Page)>
`;
